tests/ui/static/issue-18118-2.rs

```diff
@@ -1,6 +1,8 @@
+//@ check-pass
+
 pub fn main() {
-    const z: &'static isize = {
-        static p: isize = 3;
-        &p //~ ERROR referencing statics
+    const Z: &'static isize = {
+        static P: isize = 3;
```

Comment:
Same here, making this `static mut` likely better preserves the original intent of the test.

Reply:
Switched to `static mut`.

Reply:
This probably needs the `unsafe` block as well, then.

Reply:
I actually am not sure this test has such a purpose, the original test this was forked off from was originally this:
```rust
pub fn main() {
    static z: &'static int = {
        let p = 3;
        &p
    };
}
```

which caught an ICE case, and then there was another case, but I think we do in fact want this to be check-pass now.

Reply:
whichever direction we go, for this one, we have to change the meaning of the test a bit in order to keep it, so we should note down what we're testing for now.

Reply:
I don't think it's worth doing a whole lot if history digging here, that way exceeds what we usually expect from PRs like this. But I'm wary of changing a "fail" test into a "pass" test, hence the suggestions to use `static mut` + `unsafe` to keep the test vaguely like how it was before.

Reply:
I mean, I would agree except I already did the digging! :^)

Reply:
I am happy to defer to your preference, tho'.

Reply:
> then this forked test was introduced when constness checking was reworked, where there is an inner static that is then borrowed instead of a local. So I think we do in fact want this to be check-pass now because the forked test, in a sense, is in fact something we simply want to work now.

It is not clear to me what the motivation for the forked test is -- is it to ensure that we also don't ICE for this other kind of failure, or to ensure we don't ICE if/when this gets accepted? Could be either.

TBH I'd also be fine with removing the test, it doesn't seem to still have much of a purpose. :shrug:

Reply:
Yeah, I'm saying that unless @oli-obk pops up and happens to remember what was going on  in commit https://github.com/rust-lang/rust/blob/1471d932a9d191216c0da318c085b508f9bdfc1f/src/test/compile-fail/issue-18118-2.rs differently than what I am saying, it was basically just doing error validation on this alternate case because const evaluation was being reordered before typechecking.

Reply:
this was https://github.com/rust-lang/rust/pull/30930 to address https://github.com/rust-lang/rust/issues/30887

Reply:
in the absence of "this should error"... because it is now accepted... it does seem basically pointless except as a pass-test. of which it is probablyduplicated five times over.

Reply:
>  it does seem basically pointless except as a pass-test. of which it is probablyduplicated five times over.

So let's just remove it then.